magento/magento2
# Set correct Content-Type header for ajax responses in the Magento_UI module.

### Description
We started noticing issues when New Relic's Browser profiling was enabled, which inserts a javascript block into `text/html` responses whenever a `<head>` tag was found in such a response.

This has the possibility to break certain ajax responses in Magento's backend, since they have Content-Type: `text/html` instead of `application/json`.

I'm not sure if this fix is the correct fix, and if there aren't other violations in the code for this same problem. Feel free to let me know if that should be the case and I'll try to add them to the PR.

### Fixed Issues (if relevant)
None that I could find

### Manual testing scenarios
1. Tested on Magento CE 2.1.7 and develop branch (rev: 45896ba)
2. Make sure you have a New Relic integration setup with your server and that the Browser Profiling is enabled.
3. Edit or create a new CMS Page and add the following to the 'Layout Update XML' field:
```xml
<head>
    <meta name="robots" content="NOINDEX,NOFOLLOW"/>
</head>
```
4. Save the page and return to the cms pages list
5. The cms page grid no longer loads and an error: "Something went wrong" appears
<img width="1502" alt="screen shot 2017-08-13 at 17 02 23" src="https://user-images.githubusercontent.com/85479/29250916-992a7102-804b-11e7-92b9-71dc81dde7e8.png">
6. Inspecting the response, it contains invalid json, because New Relic inserted its javascript in there:

```
...
"layout_update_xml":"<head><script type="text/javascript">window.NREUM||(NREUM={}) ... </script>\r\n    <meta name=\"robots\" content=\"NOINDEX,NOFOLLOW\"\/>\r\n<\/head>"
...
```

Changing the Content-Type to `application/json` no longer makes New Relic insert its javascript into the response and the grid loads perfectly fine.

### Contribution checklist
